src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -33,6 +33,34 @@ public sealed partial class CryptoLib : NativeContract
 
         internal CryptoLib() : base() { }
 
+        /// <summary>
+        /// Recovers the public key from a secp256k1 signature in a single byte array format.
+        /// </summary>
+        /// <param name="message">The original message that was signed.</param>
+        /// <param name="signature">The 65-byte signature in format: r[32] + s[32] + v[1]. 64-bytes for eip-2098, where v must be 27 or 28.</param>
+        /// <param name="hashAlgorithm">The hash algorithm to be used hash the message.</param>
+        /// <returns>The recovered public key in compressed format, or null if recovery fails.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 10, Name = "recoverSecp256K1")]
+        public static byte[] RecoverSecp256K1(byte[] message, byte[] signature, HashAlgorithm hashAlgorithm)
```

Comment:
Hi shagon, i think what we should handle is passing the hash and signature, since the message sometimes require special format to be hashed which is not possible to be handled by our native contract, such as `keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hash))` . So we should leave the hash thing to the costum contract.

Reply:
`HasAlgoritm.None` is allowed

Reply:
Removed none